src/ssl/mod.rs

```diff
@@ -1,4 +1,4 @@
-#[allow(unused)]
+#[allow(dead_code)] // disable warning of not using the String in the Error
```

Comment:
this comment is good, but I feel it is missing an explanation of why this is necessary (instead of just removing String), I bet conditional compilation is the reason.

Reply:
This is some kind of false positive of cargo linter, it report that the String is not used by the Error struct, and it notice that you have derive Debug(and this is why we cannot remove the String, because if we remove it, then users of http_io cannot know the description of the error anymore)

Here is the error message, the linter knows you are using it, but it just will report this.
![image](https://github.com/user-attachments/assets/eaca14cb-5a04-4e0c-9cca-7b352a643f9d)